azurerm/internal/ar/client.go

```diff
@@ -10,26 +10,36 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/hashicorp/go-azure-helpers/sender"
 	"github.com/hashicorp/terraform/httpclient"
-	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 	"github.com/terraform-providers/terraform-provider-azurerm/version"
 )
 
-func ConfigureClient(client *autorest.Client, auth autorest.Authorizer, partnerId string, skipProviderReg bool) {
-	setUserAgent(client, partnerId)
-	client.Authorizer = auth
-	client.Sender = sender.BuildSender("AzureRM")
-	client.PollingDuration = 60 * time.Minute
-	client.SkipResourceProviderRegistration = skipProviderReg
-	client.RequestInspector = azure.WithCorrelationRequestID(azure.CorrelationRequestID())
+type ClientOptions struct {
+	Authorizer                 autorest.Authorizer
+	ProviderName               string
+	PartnerId                  string
+	PollingDuration            time.Duration
+	SkipProviderReg            bool
+	EnableCorrelationRequestID bool
+}
+
+func ConfigureClient(c *autorest.Client, o *ClientOptions) {
```

Comment:
since different clients use different authentication options here; this won't work as-is - instead we'll need to pass in (at least) the Authorizer to use for the particular client here

Reply:
we could probably also make this a method on the 'Options' object here, e.g.

```suggestion
func (options ClientOptions) ConfigureClient(c *autorest.Client, auth autorest.Authorizer) {
```